cmd/otel-allocator/prehook/relabel.go

```diff
@@ -93,6 +115,14 @@ func (tf *RelabelConfigTargetFilter) GetConfig() map[string][]*relabel.Config {
 	return relabelCfgCopy
 }
 
+func replaceShard(body []byte) []byte {
+	// See this thread [https://github.com/open-telemetry/opentelemetry-operator/pull/1124/files#r983145795]
+	// for why SHARD == 0 is a necessary substitution. Otherwise the keep action that uses this env variable,
+	// would not match the regex and all targets end up dropped.
+	shard := "0"
```

Comment:
nit: also should include that shard will always be 0 here, and it doesn't make sense to read the SHARD from the environment.

Reply:
Added!